README.md

```diff
@@ -8,3 +8,112 @@
 # pr-log
 
 > Changelog generator based on GitHub Pull Requests
+
+The main features:
+
+* Writes in a `CHANGELOG.md` from merged GitHub pull requests since the last tag. This works by 
+    * first getting a list of all tags
+    * than removing all tags that are not compatible to [semver versioning](http://semver.org/)
+    * sort the tags
+    * getting the git log from the last tag until now
+    * If no `CHANGELOG.md` existed, it will create the file else it will write prepending to it
+* Friendly CLI
+    * Get usage by running `pr-log --help`
+    * Error messages that help correcting usage mistakes. E.g.
+        * Missing first command line argument: `Error: version-number not specified`
+        * Local branch is outdated compared to the remote branch: `Error: Local git master branch is 0 commits ahead and 2 commits behind of origin/master`
+        * The current working directory is not clean (e.g. contains files that are modified): `Error: Local copy is not clean`
+* Well tested
+
+## Install & setup
+
+Simply run this to install `pr-log`:
+
+```
+npm install pr-log
+```
+
+Currently the categories for the changelog are hardcoded in `lib/validLabels.js`. The following categories are predefined (`<label name>: <human friendly name>)`:
+
+```
+bug: 'Bug Fixes',
+upgrade: 'Dependency Upgrades',
+documentation: 'Documentation',
+feature: 'Features',
+enhancement: 'Enhancements',
+build: 'Build-Related',
+breaking: 'Breaking Changes'
+```
+
+To fully profit from `pr-log` your GitHub project needs some small configuration:
+
+* Create the labels mentioned above (you can create GitHub labels from `Issues -> Labels -> New Label`)
+* Set the correct labels on your pull requests
+* Use correct semver versioning for your tags (e.g. `2.4.7`)
+
+To setup the categories of the project correctly that will be used for the changelog, there 
+
```

Comment:
You actually HAVE TO set exactly one of those labels per pull request.
There will be an error thrown if you use none of the valid labels or if you use more than one.